x/emissions/keeper/keeper.go

```diff
@@ -138,6 +138,9 @@ type Keeper struct {
 	// map of (topic, worker) -> inference
 	inferences collections.Map[collections.Pair[TOPIC_ID, sdk.AccAddress], state.Inference]
 
+	// map of (topic, worker) -> forecast[]
+	forecasts collections.Map[collections.Pair[TOPIC_ID, sdk.AccAddress], state.Forecast]
```

Comment:
why are we storing the same data twice in forecasts and allForecasts?

We should be able to use better data structures than this, some combination maybe triple indexed to only store data once